### PR TITLE
Update dependencies (rustc nightly-2022-08-15, viper v.22.07-release)

### PR DESCRIPTION
**Pull request by viper-admin**
* [x] Update Viper version to `v.22.07-release`.
* [x] Update rustc version to `nightly-2022-08-15`.
* [x] Run `cargo audit` and fix the issues.
* [x] Manualy update outdated dependencies (see the list below).
* [x] Manualy run `cargo update`.

<details><summary>List of direct outdated dependencies:</summary>

```
$ cargo outdated --root-deps-only --workspace

info: syncing channel updates for 'nightly-2022-08-15-x86_64-unknown-linux-gnu'
info: latest update on 2022-08-15, rust version 1.65.0-nightly (801821d15 2022-08-14)
info: downloading component 'cargo'
info: downloading component 'clippy'
info: downloading component 'llvm-tools-preview'
info: downloading component 'rust-std'
info: downloading component 'rustc'
info: downloading component 'rustc-dev'
info: downloading component 'rustfmt'
info: installing component 'cargo'
info: installing component 'clippy'
info: installing component 'llvm-tools-preview'
info: installing component 'rust-std'
info: installing component 'rustc'
info: installing component 'rustc-dev'
info: installing component 'rustfmt'
    Updating git repository `https://github.com/rust-lang/cargo.git`
analysis
================
Name        Project  Compat  Latest   Kind    Platform
----        -------  ------  ------   ----    --------
serde       1.0.141  ---     1.0.143  Normal  ---
serde_json  1.0.82   ---     1.0.83   Normal  ---

prusti
================
Name    Project  Compat  Latest  Kind   Platform
----    -------  ------  ------  ----   --------
chrono  0.4.19   ---     0.4.22  Build  ---

prusti-common
================
Name   Project  Compat  Latest   Kind    Platform
----   -------  ------  ------   ----    --------
serde  1.0.141  ---     1.0.143  Normal  ---

prusti-launch
================
Name   Project  Compat  Latest   Kind    Platform
----   -------  ------  ------   ----    --------
ctrlc  3.2.2    ---     3.2.3    Normal  ---
nix    0.24.2   ---     0.25.0   Normal  cfg(unix)
serde  1.0.141  ---     1.0.143  Normal  ---

viper
================
Name     Project  Compat  Latest   Kind    Platform
----     -------  ------  ------   ----    --------
futures  0.3.21   ---     0.3.23   Normal  ---
serde    1.0.141  ---     1.0.143  Normal  ---

vir
================
Name   Project  Compat  Latest   Kind    Platform
----   -------  ------  ------   ----    --------
serde  1.0.141  ---     1.0.143  Normal  ---

prusti-specs
================
Name        Project  Compat  Latest   Kind    Platform
----        -------  ------  ------   ----    --------
serde       1.0.141  ---     1.0.143  Normal  ---
serde_json  1.0.82   ---     1.0.83   Normal  ---

prusti-interface
================
Name   Project  Compat  Latest   Kind    Platform
----   -------  ------  ------   ----    --------
serde  1.0.141  ---     1.0.143  Normal  ---

prusti-viper
================
Name   Project  Compat  Latest   Kind    Platform
----   -------  ------  ------   ----    --------
serde  1.0.141  ---     1.0.143  Normal  ---

prusti-server
================
Name   Project  Compat  Latest   Kind    Platform
----   -------  ------  ------   ----    --------
clap   3.2.16   ---     3.2.17   Normal  ---
serde  1.0.141  ---     1.0.143  Normal  ---

prusti-smt-solver
================
Name     Project  Compat  Latest  Kind    Platform
----     -------  ------  ------  ----    --------
futures  0.3.21   ---     0.3.23  Normal  ---

test-crates
================
Name   Project  Compat  Latest   Kind    Platform
----   -------  ------  ------   ----    --------
clap   3.2.16   ---     3.2.17   Normal  ---
serde  1.0.141  ---     1.0.143  Normal  ---
```
</details>

@vakaras could you take care of this?

### CODE REVIEW

**Comment by vakaras** [comment]
@Aurel300 It seems that the test fails because the refinement code does not find the trait precondition. Do you know what could be the root cause of the problem, or at least where to look for it?

File: prusti-interface/src/utils.rs [hunk 1]
```diff
@@ -210,22 +210,24 @@ pub fn get_attributes(tcx: ty::TyCtxt<'_>, def_id: DefId) -> &[prusti_rustc_inte
...
+        ast::AttrKind::Normal(normal_attr) => {
+            match &normal_attr.item {
+                ast::AttrItem {
+                    path:
+                        ast::Path {
+                            span: _,
+                            segments,
+                            tokens: _,
+                        },
+                    args: ast::MacArgs::Empty,
```
Comment (Aurel300) [inline prusti-interface/src/utils.rs]: @vakaras This broke it, some of our attributes are `::Eq` :)

I'll push the fix + update to today's toolchain.

File: analysis/src/bin/analysis-driver.rs [hunk 2]
```diff
@@ -61,13 +61,13 @@ fn get_attribute<'tcx>(
                         },
                     args: ast::MacArgs::Empty,
```
Comment (Aurel300) [inline analysis/src/bin/analysis-driver.rs:62]: This has been here for a while, so the `analysis` crate only sees `prusti` attributes without arguments/values. Is this intended, or even needed @fpoli ?

File: prusti-interface/src/utils.rs (hunk 1 above)
Comment (vakaras) [reply prusti-interface/src/utils.rs]: Oh, thank for finding this out!

**Pull request merged**